enterprise/coderd/prebuilds/claim_test.go

```diff
@@ -0,0 +1,567 @@
+				Options: &coderdtest.Options{
+					IncludeProvisionerDaemon: true,
+					Database:                 spy,
+					Pubsub:                   pubsub,
+				},
+
+				EntitlementsUpdateInterval: time.Second,
+			})
+
+			reconciler := prebuilds.NewStoreReconciler(spy, pubsub, codersdk.PrebuildsConfig{}, logger, quartz.NewMock(t))
+			var claimer agplprebuilds.Claimer = prebuilds.NewEnterpriseClaimer(spy)
+			api.AGPL.PrebuildsClaimer.Store(&claimer)
+
+			version := coderdtest.CreateTemplateVersion(t, client, owner.OrganizationID, templateWithAgentAndPresetsWithPrebuilds(desiredInstances))
+			_ = coderdtest.AwaitTemplateVersionJobCompleted(t, client, version.ID)
+			coderdtest.CreateTemplate(t, client, owner.OrganizationID, version.ID)
+			presets, err := client.TemplateVersionPresets(ctx, version.ID)
+			require.NoError(t, err)
+			require.Len(t, presets, presetCount)
+
+			userClient, user := coderdtest.CreateAnotherUser(t, client, owner.OrganizationID, rbac.RoleMember())
+
+			// Given: the reconciliation state is snapshot.
+			state, err := reconciler.SnapshotState(ctx, spy)
+			require.NoError(t, err)
+			require.Len(t, state.Presets, presetCount)
+
+			// When: a reconciliation is setup for each preset.
+			for _, preset := range presets {
+				ps, err := state.FilterByPreset(preset.ID)
+				require.NoError(t, err)
+				require.NotNil(t, ps)
+				actions, err := reconciler.CalculateActions(ctx, *ps)
+				require.NoError(t, err)
+				require.NotNil(t, actions)
+
+				require.NoError(t, reconciler.ReconcilePreset(ctx, *ps))
+			}
+
+			// Given: a set of running, eligible prebuilds eventually starts up.
+			runningPrebuilds := make(map[uuid.UUID]database.GetRunningPrebuiltWorkspacesRow, desiredInstances*presetCount)
+			require.Eventually(t, func() bool {
+				rows, err := spy.GetRunningPrebuiltWorkspaces(ctx)
+				if err != nil {
+					return false
+				}
+
+				for _, row := range rows {
+					runningPrebuilds[row.CurrentPresetID.UUID] = row
+
+					if !tc.markPrebuildsClaimable {
+						continue
+					}
+
+					agents, err := db.GetWorkspaceAgentsInLatestBuildByWorkspaceID(ctx, row.ID)
+					if err != nil {
+						return false
+					}
+
+					// Workspaces are eligible once its agent is marked "ready".
+					for _, agent := range agents {
+						err = db.UpdateWorkspaceAgentLifecycleStateByID(ctx, database.UpdateWorkspaceAgentLifecycleStateByIDParams{
+							ID:             agent.ID,
+							LifecycleState: database.WorkspaceAgentLifecycleStateReady,
+							StartedAt:      sql.NullTime{Time: time.Now().Add(time.Hour), Valid: true},
+							ReadyAt:        sql.NullTime{Time: time.Now().Add(-1 * time.Hour), Valid: true},
+						})
+						if err != nil {
+							return false
+						}
+					}
+				}
+
+				t.Logf("found %d running prebuilds so far, want %d", len(runningPrebuilds), expectedPrebuildsCount)
+
+				return len(runningPrebuilds) == expectedPrebuildsCount
+			}, testutil.WaitSuperLong, testutil.IntervalSlow)
```

Comment:
I'm wary about you having `testutil.WaitSuperLong` here when you're using `testutil.WaitMedium` for the `context.Context` you define above.

`testutil.Eventually` is a context-aware version of `require.Eventually` that I think could be useful here.